src/vcpkg/binarycaching.cpp

```diff
@@ -2332,24 +2352,28 @@ ExpectedS<std::vector<std::unique_ptr<IBinaryProvider>>> vcpkg::create_binary_pr
     std::vector<std::unique_ptr<IBinaryProvider>> providers;
     if (!s.gcs_read_prefixes.empty() || !s.gcs_write_prefixes.empty())
     {
+        LockGuardPtr<Metrics>(g_metrics)->track_property("binarycaching-gcs", "defined");
```

Comment:
>After thinking about it, I believe we should collect metrics after the binary providers have been instantiated.

The problem is that we have different providers that get smashed into the same "provider" by the parser but want different telemetry events for each.

>Since users can override binary sources via the command line, for example:

`clear;` and `clear()` should be making sure this use case works. Seems like a missing test.